feature-libs/order/core/facade/order-history-extended-view.service.ts

```diff
@@ -0,0 +1,260 @@
+      switchMap((order: Order) => {
+        //-----------------> filling consignment tracking
+        let orderView: OrderView = { ...order };
+        orderView.consignments = [];
+        const requests = (order.consignments ?? []).map(
+          (consignment: Consignment) => {
+            let consignmentView: ConsignmentView = { ...consignment };
+            if (consignment.code && consignment.trackingID) {
+              return this.getConsignmentTracking(
+                order?.code ?? '',
+                consignment.code
+              ).pipe(
+                map((trackingInfo: ConsignmentTracking) => {
+                  consignmentView.consignmentTracking = trackingInfo;
+                  orderView.consignments?.push(consignmentView);
+                  return orderView;
+                })
+              );
+            } else {
+              orderView.consignments?.push(consignmentView);
+              return of(orderView);
+            }
+          }
+        );
+        if (requests === undefined || requests.length < 1) {
+          return of(orderView);
+        }
+        return combineLatest(requests).pipe(
+          switchMap((orders: OrderView[] | undefined) => {
+            if (orders !== undefined) {
+              return of(orders[0]);
+            } else {
+              return of(order);
+            }
+          })
+        );
+        //<-----------------
+      })
+    );
+  }
+
+  getOrderListWithDetails(
+    pageSize: number
+  ): Observable<OrderHistoryListView | undefined> {
+    let orderListView: OrderHistoryListView = {};
+    return this.orderHitsoryService.getOrderHistoryList(pageSize).pipe(
+      switchMap((orderList: OrderHistoryList | undefined) => {
+        orderListView.pagination = orderList?.pagination;
+        orderListView.sorts = orderList?.sorts;
+        orderListView.orders = [];
+        const requests = (orderList?.orders ?? []).map(
+          (order: OrderHistory) => {
+            let orderView: OrderHistoryView = { ...order };
+            return this.getOrderDetailsWithTracking(order?.code ?? '').pipe(
+              map((orderDetail: OrderView | undefined) => {
+                /** filling extra fields ---> */
+                orderView.entries = orderDetail?.entries;
+                orderView.consignments = orderDetail?.consignments;
+                orderView.unconsignedEntries = orderDetail?.unconsignedEntries;
+                orderView.returnRequests = [];
+                /** filling extra fields <--- */
+                orderListView.orders?.push(orderView);
+                return orderListView;
+              })
+            );
+          }
+        );
+        return combineLatest(requests);
+      }),
+      map((requests: OrderHistoryListView[] | undefined) => {
+        if (requests !== undefined) {
+          return requests[0];
+        } else {
+          return {};
+        }
+      })
+    );
+  }
+
+  getOrderHistoryList(
```

Comment:
[MEDIUM]
Method names in this service sometimes say `order` and sometimes `orderHistory`. Let's make them consistent. e.g:
`getOrderHistoryList` - history
`getOrderListWithDetails` - no history

I don't mind dropping the word `history` or keeping it. Just let's make it consistent

Reply:
done